dvc/objects/db/base.py

```diff
@@ -140,9 +167,16 @@ def check(self, hash_info, check_hash=True):
         - Remove the file from cache if it doesn't match the actual hash
         """
 
+        if check_staging and self.staging is not None:
```

Comment:
Feels a bit strange that the odb itself knows that it is for staging. Feels like this needs to be checked somewhere on a higher level.

Reply:
Going back to the mirror concept, it seemed like this still gets tied to the parent ODB to me.